tests/foreman/api/test_variables.py

```diff
@@ -15,16 +15,51 @@
 
 @Upstream: No
 """
-
-from robottelo.decorators import run_only_on, stubbed, tier1, tier2, tier3
+from random import choice
+
+from fauxfactory import gen_integer, gen_string
+from nailgun import entities
+from requests import HTTPError
+
+from robottelo import ssh
+from robottelo.config import settings
+from robottelo.datafactory import invalid_values_list, valid_data_list
+from robottelo.decorators import (
+    run_only_on,
+    skip_if_bug_open,
+    stubbed,
+    tier1,
+    tier2,
+)
 from robottelo.test import APITestCase
 
 
 class SmartVariablesTestCase(APITestCase):
     """Implements Smart Variables tests in API"""
 
+    @classmethod
+    def setUpClass(cls):
+        """Import some parametrized puppet classes. This is required to make
+        sure that we have data to be able to perform interactions with smart
+        class variables.
+        """
+        super(SmartVariablesTestCase, cls).setUpClass()
+        cls.host_name = settings.server.hostname
+        ssh.command('puppet module install --force puppetlabs/ntp')
+        cls.env = entities.Environment().search(
+            query={'search': 'name="production"'}
+        )
+        if len(cls.env) < 0:
```

Comment:
Would it be len(cls.env) == 0? If env is a list it will never have negative len.